pkg/types/vsphere/validation/machinepool.go

```diff
@@ -18,25 +26,29 @@ func ValidateMachinePool(platform *vsphere.Platform, machinePool *types.MachineP
 	if vspherePool.MemoryMiB < 0 {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("memoryMB"), vspherePool.MemoryMiB, "memory size must be positive"))
 	}
-	if vspherePool.NumCPUs < 0 {
+	numCPUs := vspherePool.NumCPUs
+	if numCPUs < 0 {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("cpus"), vspherePool.NumCPUs, "number of CPUs must be positive"))
 	}
-	if vspherePool.NumCoresPerSocket < 0 {
+	numCoresPerSocket := vspherePool.NumCoresPerSocket
+	if numCoresPerSocket < 0 {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("coresPerSocket"), vspherePool.NumCoresPerSocket, "cores per socket must be positive"))
 	}
 
-	defaultCoresPerSocket := int32(4)
-	defaultNumCPUs := int32(4)
-	if vspherePool.NumCPUs > 0 {
-		if vspherePool.NumCoresPerSocket > vspherePool.NumCPUs {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("coresPerSocket"), vspherePool.NumCoresPerSocket, "cores per socket must be less than number of CPUs"))
-		} else if vspherePool.NumCoresPerSocket > 0 && vspherePool.NumCPUs%vspherePool.NumCoresPerSocket != 0 {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("cpus"), vspherePool.NumCPUs, "numCPUs specified should be a multiple of cores per socket"))
-		} else if vspherePool.NumCPUs%defaultCoresPerSocket != 0 {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("cpus"), vspherePool.NumCPUs, "numCPUs specified should be a multiple of cores per socket which is by default 4"))
-		}
-	} else if vspherePool.NumCoresPerSocket > defaultNumCPUs {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("coresPerSocket"), vspherePool.NumCoresPerSocket, "cores per socket must be less than number of CPUs which is by default 4"))
+	// Either number set by user or a default value
+	if numCPUs <= 0 {
+		numCPUs = defaultNumCPUs
+	}
+	if numCoresPerSocket <= 0 {
+		numCoresPerSocket = defaultCoresPerSocket
+	}
```

Comment:
Info message when a default value is in use esp when one value is specified by the user and the other is not.

Reply:
If you feel strongly about that, open a bug so we can fix in master first and then backport all the way.

Reply:
+1 I see that this is a back-port.